src/main/java/com/amadeus/jenkins/plugins/workflow/libs/HttpRetriever.java

```diff
@@ -226,9 +226,18 @@ private void doRetrieve(String sourceURL, String name, String version, FilePath
       versionMessage += "From HTTP URL: " + sourceURL;
       listener.getLogger().println(versionMessage);
 
+      // check to see if single directory is present, possibly encompassing the whole shared library that is configured
+      if (lease.path.list().size() == 1 && lease.path.list().get(0).isDirectory()) {
+          // perform a final check to make sure the upper level directory isn't just a single dir of a shared library,
+          // (e.g.`src`, `vars`, or `resources`) before moving the child contents into the proper FS location
+          String sharedLibUpperDir = lease.path.list().get(0).getName();
+          if (!sharedLibUpperDir.equals("src") && !sharedLibUpperDir.equals("vars") && !sharedLibUpperDir.equals("resources")) {
```

Comment:
For the sake of completeness, could you also add `resources` since it is mentioned in the official [documentation](https://www.jenkins.io/doc/book/pipeline/shared-libraries/)? I doubt there could be someone that packages a library with just resources... But just in case